### PR TITLE
HW_1

### PR DESCRIPTION
**Pull request by andrylviv**

### CODE REVIEW

**Comment by KostyantynPanchenko** [comment]
The task itself is completed without any notes. But there are a lot of fixes that should be done. Most of them are related to the code style violations and some to the design. Please fix according to the comments and check ALL classes (not only commented) with regards to the code style and readability.

References:
* Java code conventions - https://www.oracle.com/java/technologies/javase/codeconventions-contents.html
* Google code style (widely used) - https://google.github.io/styleguide/javaguide.html

Please use formatting options of your IDE (`Ctrl + Alt + Shift + o` if you are an Intellij IDEA user)

**Comment by KostyantynPanchenko** [comment]
Recommendation for commit messages - I would go with the following template HW_{homework_number} - description of what was done.

Simplest example would be:
`HW_1 - implement items 1-5`
`HW_1 - implement items 6-9`
`HW_1 - implement item 10`

More verbose and understandable:
`HW_1 - implement POJO classes and base configuration class`
`HW_1 - provide implementations for Animal interface and extend configuration with PetConfig`
`HW_1 - add more options with Primary and Qualifier annotations`

**Review by KostyantynPanchenko — CHANGES_REQUESTED** [review]
Please pay attention to missed comments and fix:
* `PetConfig` - please remove extra blank line at the end of the class, only one blank line is required (got two)
* `OtherBeanA` - operators should be separated with a space from both sides. Example: `this.getClass().getSimpleName()+". "+ beanA.getClass().getSimpleName()+ was ...` should be `this.getClass().getSimpleName() + ". " + beanA.getClass().getSimpleName() + was ...`. Please indent properly ALL operators
* `OtherBeanB` - The same as for `OtherBeanA`. Plus there is a wrong message in a `toString()` - it should be `through the setter` instead of `through the constructor`
* `Pet`, line 14 - add space after colon

File: src/main/java/com/epam/spring/homework1/other/OtherBeanA.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.epam.spring.homework1.other;
+
+import com.epam.spring.homework1.beans.BeanA;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OtherBeanA {
+    BeanA beanA;
```
[the remark this answers is not in the document]
Comment (andrylviv) [reply src/main/java/com/epam/spring/homework1/other/OtherBeanA.java]: was fixed in "HW_1 - violation fixes" commit

File: src/main/java/com/epam/spring/homework1/other/OtherBeanB.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.epam.spring.homework1.other;
+
+import com.epam.spring.homework1.beans.BeanB;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OtherBeanB {
+    BeanB beanB;
```
[the remark this answers is not in the document]
Comment (andrylviv) [reply src/main/java/com/epam/spring/homework1/other/OtherBeanB.java]: was fixed in "HW_1 - violation fixes" commit

**Review by KostyantynPanchenko — APPROVED** [review]

**Pull request closed without merging**